src/commands/test/lib/xcuitest-preparer.ts

```diff
@@ -0,0 +1,93 @@
+import * as fs from "fs";
+import * as iba from "../../../util/misc/ios-bundle-archiver";
+import * as path from "path";
+import * as pfs from "../../../util/misc/promisfied-fs";
+import * as pglob from "../../../util/misc/promisfied-glob";
+import { TestCloudError } from "./test-cloud-error";
+
+export class XCUITestPreparer {
+  private readonly artifactsDir: string;
+  private readonly buildDir: string;
+  private testIpaPath: string;
+
+  constructor(artifactsDir: string, buildDir: string, testIpaPath: string) {
+    if (!artifactsDir) {
+      throw new Error("Argument artifactsDir is required");
+    }
+
+    if (!(buildDir || testIpaPath)) {
+      throw new Error("Either buildDir or testIpaPath argument is required");
+    }
+
+    if (buildDir && testIpaPath) {
+      throw new Error("Arguments buildDir and testIpaPath cannot be used together");
+    }
+
+    this.artifactsDir = artifactsDir;
+    this.buildDir = buildDir;
+    this.testIpaPath = testIpaPath;
+  }
+
+  public async prepare(): Promise<string> {
+    if (!pfs.exists(this.artifactsDir)) {
```

Comment:
C# compiler is much better at detecting tasks that are not awaited. Also, the type checking would prevent you from converting Task / Promise to Boolean.